news-bundle/src/EventListener/BreadcrumbListener.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\EventListener;
+
+use Contao\Config;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\Input;
+use Contao\News;
+use Contao\NewsArchiveModel;
+use Contao\NewsModel;
+use Contao\PageModel;
+use Contao\StringUtil;
+
+final class BreadcrumbListener
+{
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    public function __construct(ContaoFrameworkInterface $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    public function onGenerateBreadcrumb(array $items): array
+    {
+        $newsAlias = $this->getNewsAlias();
+        if (!$newsAlias) {
+            return $items;
+        }
+
+        $newsArchive = $this->getNewsArchive();
+        if (!$newsArchive) {
+            return $items;
+        }
+
+        $news = $this->getNews($newsAlias, $newsArchive);
+        if (!$news) {
+            return $items;
+        }
+
+        if ($newsArchive::BREADCRUMB_MODE_EXTEND === $newsArchive->breadcrumbMode) {
+            return $this->addBreadcrumbItem($items, $news);
+        }
+
+        return $this->overrideActiveBreadcrumbItem($items, $news);
+    }
+
+    private function getNewsAlias(): ?string
+    {
+        if (!isset($GLOBALS['objPage'])) {
+            return null;
+        }
+
+        $inputAdapter = $this->framework->getAdapter(Input::class);
+        $configAdapter = $this->framework->getAdapter(Config::class);
+
+        if ($configAdapter->__call('get', ['useAutoItem'])) {
```

Comment:
Is there a specific reason you are using `->__call('get', […])` insteadof just `->get(…)`?

```suggestion
        if ($configAdapter->get('useAutoItem')) {
```

Reply:
It's just the style I personally prefer because it's the best way to cope IDE type detection and inspections without decreasing inspection level (Yes, even using `/** @var \Contao\Config $configAdapter*/` would trigger an inspection notice). Since I don't have to maintain it, I don't mind changing it to the style you prefer.

Reply:
@ausi Okay, I changed all adapter methods calls being magic now.